src/python/grpcio_reflection/grpc/reflection/v1alpha/reflection.py

```diff
@@ -0,0 +1,142 @@
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Reference implementation for reflection in gRPC Python."""
+
+import threading
+
+import grpc
+from google.protobuf import descriptor_pb2
+from google.protobuf import descriptor_pool
+
+from grpc.reflection.v1alpha import reflection_pb2
+
+_POOL = descriptor_pool.Default()
+
+def _not_found_error():
+  return reflection_pb2.ServerReflectionResponse(
+      error_response=reflection_pb2.ErrorResponse(
+          error_code=grpc.StatusCode.NOT_FOUND.value[0],
+          error_message=grpc.StatusCode.NOT_FOUND.value[1].encode(),
+      )
+  )
+
+def _file_descriptor_response(descriptor):
+  proto = descriptor_pb2.FileDescriptorProto()
+  descriptor.CopyToProto(proto)
+  serialized_proto = proto.SerializeToString()
+  return reflection_pb2.ServerReflectionResponse(
+      file_descriptor_response=reflection_pb2.FileDescriptorResponse(
+        file_descriptor_proto=(serialized_proto,)
+      ),
+  )
+
+
+class ReflectionServicer(reflection_pb2.ServerReflectionServicer):
+  """Servicer handling RPCs for service statuses."""
+
+  def __init__(self, service_names, pool=_POOL):
+    """Constructor.
+
+    Args:
+      service_names: Iterable of fully-qualified service names available.
+    """
+    self._service_names = list(service_names)
+    self._pool = pool
+
+  def _file_by_filename(self, filename):
+    try:
+      descriptor = self._pool.FindFileByName(filename)
+    except KeyError:
+      return _not_found_error()
+    return _file_descriptor_response(descriptor)
```

Comment:
Since this statement is mutually exclusive with the `return` statement in the `except:` block, tuck it into an `else:` to make the mutual exclusivity more visually apparent.

(Here and in the next method.)